lib/board.rb

```diff
@@ -0,0 +1,51 @@
+
+
+class Board
+  
+  attr_accessor :cells
+  
+  def initialize
+    @cells = [" "] * 9
+  end
+  
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+  
+  def position(input)
+    ix = input.to_i - 1
+    @cells[ix]
+  end
+  
+  def taken?(location)
+    loc = position(location)
+    loc == "X" || loc == "O"
+  end
+  
+  def valid_move?(index)
+    index.to_i.between?(1,9) && !taken?(index)
+  end
+  
+  def full?
+    ! (@cells.any?{|cell| cell == " " })
+  end
+  
+  def reset!
+    @cells = [" "] * 9
+  end
+  
+  def turn_count
+    moves = @cells.select{|i| i == "X" || i == "O"}
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```